### ISSUE TITLE
Date range search and filtering not working using PHP client

### ISSUE THREAD

**Issue by flashcma** [elastic/elasticsearch-php#786]
Using the kibana interface elasticsearch returns the correct data results for date range searching and filtering using a timestamp  however when the query is executed using the elasticsearch php client either version 5 or 6 the result set is empty. Note that the range search using either of the php clients and in kibana works on other fields but when the range is changed to a date field as per a working query using kibana the result set is always empty.

The following query is working in kibana and returing 1000 results
GET index/_search
{
  "size":"1000",
  "query": {
    "bool": {
      "must": [ 
           { "range": { "updateDate": { "gte": "2018-07-18T00:00:00"}}}
        ]
     }
  }
} 

However the following DOES NOT WORK using the client and returns only an empty set

$query = '
  "size":"1000",
  "query": {
    "bool": {
      "must": [ 
           { "range": { "updateDate": { "gte": "2018-07-18T00:00:00"}}}
        ]
     }
  }
 ';
 $params = [
           'index' =>ES_INDEX,
            'type' => ES_TYPE,
            'body' =>  $query ,
            'client' => [ 'ignore' => [400, 404] ]
   ];
$response = $client->search($params);

I have also tried several variations of the search using 'From' etc. and the result is the same - The searches execute perfectly using kibana but are all empty using the php client.

Background 
elastic - hosted through elastic cloud
server os: ubuntu 16.04
php version 7.1.3

**Comment by polyfractal** [comment]
If that snippet is actually what's being run, the `$body` string is not valid JSON.   It's missing the opening and closing bracket.  Normally that would throw an error, but since you're ignoring 400's in the request, the exception is probably being swallowed.

Can you try again without the ignore on `400`, and adding the opening/closing bracket so the JSON is valid?

Also note that your Kibana query does not include a type (`/index/_search`) while the PHP request specifies both an index and a type, which may be returning different results.

If that still doesn't work, can you show the result of `print_r($params)` so I can see what the full request looks like before it is sent to the cluster?

**Comment by flashcma** [comment]
Thank you for the fast reply polyfractal :) Sorry for the missing brackets. That is what I get for editing code to make it more readable in this textarea.

Using kibana console the following search returns 3000+ results

GET indexName/_search
{
    "query": {
            "bool": {
                "filter": {
                    "range": {
                        "updateDate": {
                            "from": "2018-07-16T00:00:00"
               }
            }
         }
      }
   }
}


Below in my code I removed the ignore and ran the same exact query from the kibana console above. 

The parameters ES_INDEX, and ES_TYPE are constants being brought in from a config file.

Here is the code:

        $query = '{
                "query": {
                        "bool": {
                            "filter": {
                                "range": {
                                    "updateDate": {
                                        "from": "2018-07-16T00:00:00"
                           }
                        }
                     }
                  }
               }
            }';

        $params = [
            'index' => ES_INDEX,
            'type' => ES_TYPE,
            'body' =>  $query
        ];

        $response = $client->search($params);
        dd($response);

Here is the response - returns no results 

array:4 [
  "took" => 1
  "timed_out" => false
  "_shards" => array:3 [
    "total" => 5
    "successful" => 5
    "failed" => 0
  ]
  "hits" => array:3 [
    "total" => 0
    "max_score" => null
    "hits" => []
  ]
]

To make sure that the range search works on non date fields I changed the search to a non date range field then it works. Here is an example.

In kibana console the following returns 844 hits
{
    "size":"1",
    "query": {
            "bool": {
                "filter": {
                    "range": {
                        "bedsTotal": {
                            "from": "7"
               }
            }
         }
      }
   }
}


My code below for the kibana search returns 765 (not 844?)  hits
        $query = '{
                      "query": {
                            "bool": {
                                "filter": {
                                    "range": {
                                        "bedsTotal": {
                                            "from": "7"
                               }
                            }
                         }
                      }
                   }
                }';

        $params = [
            'index' => env('ES_INDEX'),
            'type' => env('ES_TYPE'),
            'body' =>  $query
        ];
        $response = $client->search($params);

So basically the exact same date range search in kibana that returns 3000+ hits returns an empty set using the client. The range search on non date fields works but with a discrepancy in the number of hits returned.

Thanks for your help with this issue!
